plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -64,4 +72,23 @@ public void onFailure(Exception e) {
         }
     }
 
+    @VisibleForTesting
+    void deleteModelChunks(String modelId, DeleteResponse deleteResponse, ActionListener<DeleteResponse> actionListener) {
+        DeleteByQueryRequest deleteModelsRequest = new DeleteByQueryRequest(ML_MODEL_INDEX);
+        deleteModelsRequest.setQuery(new TermsQueryBuilder(MODEL_ID_FIELD, modelId));
+
+        client.execute(DeleteByQueryAction.INSTANCE, deleteModelsRequest, ActionListener.wrap(r -> {
+            if (r.getBulkFailures() == null || r.getBulkFailures().size() == 0) {
+                log.info("All model chunks are deleted for model {}", modelId);
+                actionListener.onResponse(deleteResponse);
+            } else {
+                actionListener
+                    .onFailure(new OpenSearchStatusException("Failed to delete all model chunks", RestStatus.INTERNAL_SERVER_ERROR));
```

Comment:
1. If having bulk failures here, it is more helpful to throw these failures for users to understand why, as delete API is resource related. Usually users use this API when resources are not enough. If failed, they want to know why.
2. Is it possible to partially fail deleting some chunks? If yes, is there a mechanism to clean them, like garbage collection?

Reply:
1. Added a new function to return and log different error messages. AD used the same logic in error handling in this case.
2. This is possible. That's why we will add a clean up job to retry deleting again to make sure no leftovers.